activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -952,7 +952,7 @@ def optimizer_hints(*args)
     def optimizer_hints!(*args) # :nodoc:
       args.flatten!
 
-      self.optimizer_hints_values += args
+      self.optimizer_hints_values = (optimizer_hints_values + args).uniq
```

Comment:
An alternative would be to `uniq` in `Arel::SelectManager#optimizer_hints` - or make `optimizer_hints_values` a Set. Thoughts?

Reply:
`Set` sounds interesting. How would `|=` fare here?

Reply:
+1 for `|=`, we already use `|=` for multi value methods which doesn't allow duplicated values.

e.g.
https://github.com/rails/rails/blob/d3b951f326c4facd693d5e3f92cc40ccdbf6dc1b/activerecord/lib/active_record/relation/query_methods.rb#L144